core/src/test/scala/unit/kafka/coordinator/group/GroupCoordinatorTest.scala

```diff
@@ -381,6 +381,41 @@ class GroupCoordinatorTest {
     assertEquals(firstMemberId, group.allMembers.head)
   }
 
+  @Test
+  def testLeaderFailsBeforeSyncGroup(): Unit = {
```

Comment:
Not sure if this test covers the original bug, could you elaborate a bit more?

Reply:
Probably need to rename this. I'll add a comment as well. The test fails without the patch because the group is stuck in the PreparingRebalance state and does not make the transition to Empty as expected.